### ISSUE TITLE
SaveMojo: Add `docker.save.mode` option

### ISSUE THREAD

**Issue by rhuss** [fabric8io/docker-maven-plugin#1219]
### Description

As result of the discussion in https://github.com/fabric8io/docker-maven-plugin/issues/1197#issuecomment-485228687 it would be good to have an option `docker.save.mode` with two possible values:

* *daemon* to use Docker save REST call in order to get an archive from a Doc ker daemon
* *local* to pick up the Docker archive as created by `docker:build` for sending it to the Docker daemon when creating an image.

`docker.save.file` should in both cases point to the target file name under which the archive should be created.

**Comment by radamanth** [comment]
It would be really great to be able to build docker image without the need of a Docker Daemon

**Comment by rhuss** [comment]
@rawlingsj that's a different discussion, though ;-) There will be eventually support for one way or another to create an image without a Docker daemon (which gets directly pushed to a Docker registry).

In the meantime, you might want to have a look to https://github.com/GoogleContainerTools/jib which has a Maven plugin for doing exactly this.

**Comment by Beennnn** [comment]
@radamanth As discussed in https://github.com/fabric8io/docker-maven-plugin/issues/1197#issuecomment-485228687 this change will allow to create the Dockerfile without the need for a Docker Daemon.

**Comment by rhuss** [comment]
@Beennnn tbh, #1197 is for creating the _input_ for a Docker build, but not the directly usable image (e.g. where `RUN` directives have been performed). This is a big difference.

**Comment by Beennnn** [comment]
@rhuss exactly !

**Comment by Beennnn** [comment]
@rhuss 
This issue has not changed during the past 3 months.
How can we move on to implementation ?
Can we include #1197 as a first step ?

**Comment by rhuss** [comment]
@Beennnn so sorry and big apologies for this delay. I'm recently eaten up by other works (not even Java anymore these days) but will try now to move the PRs forward.

**Comment by rhuss** [comment]
I agree, that we can split up this issue in two: One for docker:save and one for docker:build. 

As docker:build is more important to you, I suggest to implement as described in https://github.com/fabric8io/docker-maven-plugin/issues/1197#issuecomment-482028662 and then move on to this issue.

I for sure don't have the spare cycles to implement this feature, but I'm happy to help to integrate it (reviewing + merging).

So if you feel fancy, please send a PR.
